app/shared/src/commonMain/kotlin/ui/settings/SettingsPage.kt

```diff
@@ -76,8 +85,18 @@ fun SettingsPage(
             )
         }
     ) { topBarPaddings ->
-        val pagerState =
-            rememberPagerState(initialPage = initialTab.ordinal) { SettingsTab.entries.size }
+        val pageCount by remember(isDebugEnabled) {
+            mutableStateOf(if (isDebugEnabled) {
+                { SettingsTab.entries.size }
+            } else {
+                { SettingsTab.entries.size - 1 }
+            })
+        }
+        val pagerState = rememberPagerState(
+            initialPage = initialTab.ordinal,
+            pageCount = pageCount
+        )
```

Comment:
```suggestion
        val pageCount by remember {
            derivedStateOf {
                if (vm.debugSettings.enabled) SettingsTab.entries.size else SettingsTab.entries.size - 1
            }
        }
        val pagerState = rememberPagerState(
            initialPage = initialTab.ordinal,
            pageCount = { pageCount },
        )
```

`vm.debugSettings` is a state read so `derivedStateOf` can listen on updates.

No lambda object creation, no recomposition